packages/core/src/credential/Credential.ts

```diff
@@ -226,19 +226,19 @@ export function verifyDataIntegrity(input: ICredential): boolean {
  *
  */
 export function verifyDataStructure(input: ICredential): void {
-  if (!input.claim) {
+  if (!('claim' in input)) {
     throw new SDKErrors.ClaimMissingError()
   } else {
     Claim.verifyDataStructure(input.claim)
   }
   if (!input.claim.owner) {
     throw new SDKErrors.OwnerMissingError()
   }
-  if (!input.legitimations && !Array.isArray(input.legitimations)) {
+  if (!Array.isArray(input.legitimations)) {
     throw new SDKErrors.LegitimationsMissingError()
   }
 
-  if (!input.claimNonceMap) {
+  if (!('claimNonceMap' in input)) {
```

Comment:
Not sure about all these `in` checks. `in` returns true, if the property is there, but set to undefined, which is possible in some of our types.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#using_in_with_deleted_or_undefined_properties

What are your thoughts on this?

Reply:
Function expects ICredential, not e.g. Partial<ICredential>, so it is technically correct to ignore the potential undefined properties. There is a bigger question on if we should change the verify functions to rather be type guards, or at least expect an input with potential undefined properties.